packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```diff
@@ -0,0 +1,440 @@
+    MessageValidator.validateSync(jsonLdCredential)
+
+    // FIXME: SignCredentialOptions doesn't follow RFC0593
+    const attachment = this.getFormatData(jsonLdFormat, format.attachId)
+
+    return { format, attachment }
+  }
+
+  public async processRequest(agentContext: AgentContext, { attachment }: FormatProcessOptions): Promise<void> {
+    // FIXME: SignCredentialOptions doesn't follow RFC0593
+    const requestJson = attachment.getDataAsJson<SignCredentialOptionsRFC0593>()
+
+    if (!requestJson) {
+      throw new AriesFrameworkError('Missing jsonld credential request data payload')
+    }
+
+    // FIXME: validating an interface doesn't work.
+    const jsonLdCredential = new JsonLdCredential(requestJson)
+    MessageValidator.validateSync(jsonLdCredential)
+  }
+
+  public async acceptRequest(
+    agentContext: AgentContext,
+    { credentialFormats, attachId, requestAttachment }: FormatAcceptRequestOptions<JsonLdCredentialFormat>
+  ): Promise<FormatCreateReturn> {
+    const jsonLdFormat = credentialFormats?.jsonld
+
+    // sign credential here. credential to be signed is received as the request attachment
+    // (attachment in the request message from holder to issuer)
+    const credentialRequest = requestAttachment.getDataAsJson<SignCredentialOptionsRFC0593>()
+
+    let verificationMethod = credentialFormats?.jsonld?.verificationMethod
+
+    // FIXME: Need to transform from json to class instance
+    // FIXME: SignCredentialOptions doesn't follow RFC0593
+    // FIXME: Add validation of the jsonLdFormat data (if present)
+    const credentialData = jsonLdFormat ?? credentialRequest
+
+    // FIXME: we're not using all properties from the interface. If we're not using them,
+    // they shouldn't be in the interface.
+    if (!verificationMethod) {
+      verificationMethod = await this.deriveVerificationMethod(
+        agentContext,
+        credentialData.credential,
+        credentialRequest
+      )
+      if (!verificationMethod) {
+        throw new AriesFrameworkError('Missing verification method in credential data')
+      }
+    }
+    const format = new CredentialFormatSpec({
+      attachId,
+      format: JSONLD_VC,
+    })
+
+    const verifiableCredential = await this.w3cCredentialService.signCredential(agentContext, {
+      credential: JsonTransformer.fromJSON(credentialData.credential, W3cCredential),
+      proofType: credentialData.options.proofType,
+      verificationMethod: verificationMethod,
+    })
+
+    const attachment = this.getFormatData(verifiableCredential, format.attachId)
+    return { format, attachment }
+  }
+
+  /**
+   * Derive a verification method using the issuer from the given verifiable credential
+   * @param credential the verifiable credential we want to sign
+   * @return the verification method derived from this credential and its associated issuer did, keys etc.
+   */
+  public async deriveVerificationMethod(
```

Comment:
This can be private I think?

Reply:
Again, it is public because there is a test for it.

Reply:
Shouldn't make methods public just for tests. You can just call it with a ts-ignore